install_subrepos.py

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/env python3
+#
+# Copyright © Spyder Project Contributors
+# Licensed under the terms of the MIT License
+#
+
+"""
+Helper script for installing spyder and external-deps locally in editable mode.
+"""
+
+import argparse
+import os
+import re
+import sys
+from logging import Formatter, StreamHandler, getLogger
+from pathlib import Path
+from pkg_resources import DistributionNotFound, get_distribution
+from subprocess import check_output
+
+DEVPATH = Path(__file__).resolve().parent
+DEPS_PATH = DEVPATH / 'external-deps'
+BASE_COMMAND = [sys.executable, '-m', 'pip', 'install', '--no-deps']
+
+REPOS = {}
+for p in DEPS_PATH.iterdir():
+    if p.name.startswith('.'):
+        continue
+    try:
+        dist = Path(get_distribution(p.name).location)
+    except DistributionNotFound:
+        dist = None
+
+    REPOS[p.name] = {'repo': p, 'dist': dist, 'editable': p == dist}
+
+# ---- Setup logger
+fmt = Formatter('%(asctime)s [%(levelname)s] [%(name)s] -> %(message)s')
+h = StreamHandler()
+h.setFormatter(fmt)
+logger = getLogger('InstallLocal')
+logger.addHandler(h)
+logger.setLevel('INFO')
+
+
+def get_python_lsp_version():
+    """Get current version to pass it to setuptools-scm."""
+    req_file = DEVPATH / 'requirements' / 'conda.txt'
+    with open(req_file, 'r') as f:
+        for line in f:
+            if 'python-lsp-server' not in line:
+                continue
+
+            # Get version part of dependency line
+            version = line.strip().split()[1]
+
+            # Get lower bound
+            version = version.split(',')[0]
+
+            # Remove comparison signs and only leave version number
+            version = re.search(r'\d+.*', version).group()
```

Comment:
```suggestion
            version = next(iter(Requirement(line).specifier))).version
```

Use `packaging` for this instead of manual string munging.